app/src/main/java/com/concordia/mcga/utilities/pathfinding/TiledMap.java

```diff
@@ -38,12 +49,66 @@ public void setTile(int x, int y, PathFinderTile tile) {
 
     /**
      * Sets the given X,Y coordinate as walkable
+     *
      * @param indoorMapTile -
      */
     public void makeWalkable(IndoorMapTile indoorMapTile) {
         allTiles[indoorMapTile.getCoordinateX()][indoorMapTile.getCoordinateY()] = new PathFinderTile(indoorMapTile);
     }
 
+    public boolean isWalkable(IndoorMapTile indoorMapTile) {
+        return getTile(indoorMapTile.getCoordinateX(), indoorMapTile.getCoordinateY()) != null;
+    }
+
+    public PathFinderTile closestWalkable(IndoorMapTile indoorMapTile) throws MCGAPathFindingException {
+        int coordinateX = indoorMapTile.getCoordinateX();
+        int coordinateY = indoorMapTile.getCoordinateY();
+        if (isWalkable(indoorMapTile)) {
+            return allTiles[coordinateX][coordinateY];
+        } else {
+            return findBreadthFirstSearch(coordinateX, coordinateY);
+        }
+    }
+
+    /**
+     *
+     * @param coordinateX X coordinate to use for the breadth first search algorithm
+     * @param coordinateY Y coordinate to use for the breadth first search algorithm
+     * @return The closest PathFinderTile to the current given coordinates. Looks at tiles
+     *         that are directly above, below, to the left and to the right. Does not look at
+     *         tiles that are diagonal to the given coordinates.
+     * @throws MCGAPathFindingException Thrown when no valid coordinate can be found.
+     */
+    PathFinderTile findBreadthFirstSearch(int coordinateX, int coordinateY) throws MCGAPathFindingException {
+        int delta = 1;
+        int limit = Math.max(allTiles.length, allTiles[0].length);
```

Comment:
Either make this variable final or put the expression directly inside the while statement